fairlearn/reductions/_moments/error_rate.py

```diff
@@ -9,14 +9,58 @@
 from fairlearn._input_validation import _validate_and_reformat_input
 
 
+_MESSAGE_BAD_COSTS = "costs needs to be a dictionary with keys " \
+    "'fp' and 'fn' containing non-negative values, which are not both zero"
+
+
 class ErrorRate(ClassificationMoment):
-    """Misclassification error."""
+    r"""Misclassification error as a moment.
+
+    A classifier :math:`h(X)` has the misclassification error equal to
+
+    .. math::
+      P[h(X) \ne Y]
+
+    It is also possible to specify costs for false positives and false
+    negatives. The error then evaluates to
+
+    .. math::
+      c_{FP} P[h(X)=1, Y=0] + c_{FN} P[h(X)=0, Y=1]
+
+    where :math:`c_{FP}` and `c_{FN}` are the costs of false positive
+    and false negative errors respectively. The standard misclassification
+    error corresponds to :math:`c_{FP}=c_{FN}=1.0`.
+
+    Parameters
+    ----------
+    costs : dict
+        The dictionary with keys ``"fp"`` and ``"fn"`` containing the
+        costs of false positives and false negatives. If none are provided
+        costs of 1.0 are assumed.
+    """
 
     short_name = "Err"
 
+    def __init__(self, *, costs=None):
+        """Initialize the costs."""
+        super(ErrorRate, self).__init__()
+        if costs is None:
+            self.fp_cost = 1.0
+            self.fn_cost = 1.0
+        elif (type(costs) is dict
+              and costs.keys() == {'fp', 'fn'}
+              and costs['fp'] >= 0.0
+              and costs['fn'] >= 0.0
+              and costs['fp'] + costs['fn'] > 0.0):
+            self.fp_cost = costs['fp']
+            self.fn_cost = costs['fn']
+        else:
+            raise ValueError(_MESSAGE_BAD_COSTS)
+        self.sum_costs = self.fp_cost + self.fn_cost
```

Comment:
I would suggest making `sum_costs`, `fp_cost` and `fn_cost` into properties rather than fields.

Reply:
Surely the naming convention has nothing to do with whether they're fields or properties?

Reply:
So maybe I misunderstood. Why do you want them to be properties?

Reply:
Because making them properties gives a place for adding documentation (and type hints)

Reply:
Perhaps I don't remember it quite right, but I do think that changes the naming convention. Also, the costs are input args and not calculated through calling `fit` so they should not be suffixed with `_`. I like @riedgar-ms's suggestion to make them properties. You could argue that it's different for the sum of costs, but I appreciate the consistency between the three as well.
[I can't shake the feeling that I'm splitting hair here, though, so feel free to proceed either way if I don't respond further.]

Reply:
If `sum_costs` is only used in one place, removing it does make sense.